lib/ansible/executor/play_iterator.py

```diff
@@ -587,4 +588,19 @@ def _insert_tasks_into_state(self, state, task_list):
         return state
 
     def add_tasks(self, host, task_list):
-        self._host_states[host.name] = self._insert_tasks_into_state(self.get_host_state(host), task_list)
+        self.set_state_for_host(host.name, self._insert_tasks_into_state(self.get_host_state(host), task_list))
+
+    def set_state_for_host(self, hostname, state):
```

Comment:
Since this is new public API that's 3.8+, any reason we shouldn't type annotate these methods?

Reply:
+1, added.